### PR TITLE
Provide the forceVerticalCenter parameter to achieve forced vertical centering

### PR DESCRIPTION
**Pull request by linxuebin1990**
It is difficult to center the text vertically in flutter.
Therefore, according to the original idea of Android, the minimum height of the glyph is obtained by using the getBounds interface of Layout. Then through this height, adjust the vertical position of the text in the overall layout to achieve centering.

drawing

In the above figure, for each column, the left is the original and the right is the optimized.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by chinmaygarde** [comment]
Can you cross reference this with the issue you are trying to address? We aren't trying to add to the txt backend and instead focusing on the SkParagraph migration.

**Comment by jason-simmons** [comment]
Also see the `TextLeadingDistribution` API (https://api.flutter.dev/flutter/dart-ui/TextLeadingDistribution-class.html) and the discussion in https://github.com/flutter/flutter/issues/72521

**Comment by linxuebin1990** [comment]
@chinmaygarde @jason-simmons 
This submission is to solve the following problems and hope it will be helpful to the design of SkParagraph.

If there is no leading, the highest point of the text box is the ascent of the font, and the lowest point is the descent of the font.
 --------------------- ascent
    TextString
 --------------------- descent

Take NotoSansCJK-Regular (the default font for Chinese, Japanese and Korean in the Android system) font as an example, Units per Em value is 1000, ascent is 1160, descent is -320, then the vertical center point of ascent and descent is 420. Take the Chinese character "中" (Unicode 4e2d) for example, the highest point of the glyph is 837, the lowest point is -37, and the center point of the glyph itself is 380.

There is a big gap between the center points of the two. If the font size is 20 and on a device with a density of 3:
(420-380) / 1000 * 20 * 3 = 2 pixel.

**Comment by chinmaygarde** [comment]
@linxuebin1990 can you rephrase this as a feature request please? Since there is reluctance to add additional flags to this backend, I don't see this patch being accepting in its current state. But I would like to make sure the use case is supported and tested in SkParagraph.

**Comment by bird-rookie** [comment]
学习中。。

**Comment by msxenon** [comment]
Why this is not merged yet!, It solves a lot of typo-text UI issues

**Comment by flutting** [comment]
Why this is not merged yet!, It solves a lot of typo-text UI issues

**Comment by byshenliuan** [comment]
Why this is not merged yet!, It solves a lot of typo-text UI issues

**Comment by ronnieCodeWang** [comment]
Why this is not merged yet!, It solves a lot of typo-text UI issues

**Comment by HansRove** [comment]
Why this is not merged yet!, It solves a lot of typo-text UI issues
